src/components/PostDetails.tsx

```diff
@@ -2,105 +1,133 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import * as serverAction from '../api/api';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isCreateComment, setIsCreateComment] = useState(false);
+
+  useEffect(() => {
+    if (post) {
+      setIsCreateComment(false);
+      setIsLoading(true);
+
+      serverAction
+        .getComments(post.id)
+        .then(result => {
+          const results = result.map(comment => ({
+            id: comment.id,
+            postId: comment.postId,
+            name: comment.name,
+            body: comment.body,
+            email: comment.email,
+          }));
+
+          setComments(results);
+        })
+        .catch(() => setIsError(true))
```

Comment:
The error state is set when fetching comments fails, but there is no mechanism to reset this state. Consider resetting `isError` when a new post is selected or after a certain period to improve user experience.